tests/unit/test_client.py

```
@@ -1,357 +1,312 @@
+    def test_get_with_retries(self, mock_request, http_client, mock_response):
+        """Test GET request with retry logic."""
+        mock_request.side_effect = [ConnectionError("Connection failed"), mock_response]
+        result = http_client.get("https://example.com")
+        assert result == mock_response.text
+        assert mock_request.call_count == 2
+
+    @patch("requests.Session.request")
+    def test_get_max_retries_exceeded(self, mock_request, http_client):
+        """Test GET request when max retries exceeded."""
+        mock_request.side_effect = ConnectionError("Connection failed")
+        with pytest.raises(Exception):
```

Ruff (0.11.9)

Do not assert blind exception: `Exception`

(B017)